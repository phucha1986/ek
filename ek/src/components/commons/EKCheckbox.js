import React, { useState } from 'react';

const EKCheckbox = ({id, title}) => {

  const[isChecked, setIsChecked] = useState(false);
  function onToggle()
  {
    setIsChecked(!isChecked);
  }
  return (
    <div className="checkbox-field">
      <input className="checkbox__input" id={id} type="checkbox" name={id} onChange={onToggle} checked={isChecked}/>
      <label for={id} className={`checkbox__text ${isChecked ? 'icon-check' : ''}`}>{title}
      </label>
    </div> 
  );
}

export default EKCheckbox;
