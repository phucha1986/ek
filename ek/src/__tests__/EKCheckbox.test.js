import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import '@testing-library/jest-dom';
import EKCheckbox from '../components/commons/EKCheckbox';

describe('EKButton', () => {
  it('renders without prop', () => {
    const wrapper = shallow(<EKCheckbox />)
    expect(toJson(wrapper)).toMatchSnapshot()
  });
  it('renders with prop', () => {
    const props = {
      id: 'chkTest',
      title: 'test check me',
      dispatch: jest.fn()      
    }
    const wrapper = shallow(<EKCheckbox {...props}/>)
    expect(toJson(wrapper)).toMatchSnapshot()
  });
  it('should has icon-check class when checkbox is checked', () => {    
    const props = {
      id: 'chkTest',
      title: 'test check me',
      dispatch: jest.fn()      
    }
    const component = shallow(<EKCheckbox {...props}/>);  
    component.find('input#chkTest').simulate('click');
    expect(component.find('label').hasClass('icon-check')).toBeFalsy();
  });
  // it('should return false if clientSideValidation is failed', () => {
  //   const handleSubmit = jest.fn();
  //   const clientSideValidation = jest.fn(() =>  false);
  //   const props = {
  //     id: 'btnTest',
  //     title: 'click',
  //     LoadingText: 'loading',
  //     EnableLoading: false,
  //     clientSideValidation,
  //     handleSubmit,
  //     dispatch: jest.fn()      
  //   }
  //   const component = shallow(<EKButton {...props}/>);  
  //   component.find('button#btnTest').simulate('click');
  //   expect(clientSideValidation.mock.results[0].value).toBeFalsy();    
  // });
  // it('should call handleSubmit if clientSideValidation is passed', () => {
  //   const handleSubmit = jest.fn();
  //   const clientSideValidation = jest.fn(() =>  true);
  //   const props = {
  //     id: 'btnTest',
  //     title: 'click',
  //     LoadingText: 'test loading',
  //     EnableLoading: true,
  //     clientSideValidation,
  //     handleSubmit,
  //     dispatch: jest.fn()      
  //   }
  //   const component = shallow(<EKButton {...props}/>);  
  //   component.find('button#btnTest').simulate('click');
  //   expect(handleSubmit).toHaveBeenCalled();    
  // });
})