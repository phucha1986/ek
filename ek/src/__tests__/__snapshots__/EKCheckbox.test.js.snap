// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EKButton renders with prop 1`] = `
<div
  className="checkbox-field"
>
  <input
    checked={false}
    className="checkbox__input"
    id="chkTest"
    name="chkTest"
    onChange={[Function]}
    type="checkbox"
  />
  <label
    className="checkbox__text "
    for="chkTest"
  >
    test check me
  </label>
</div>
`;

exports[`EKButton renders without prop 1`] = `
<div
  className="checkbox-field"
>
  <input
    checked={false}
    className="checkbox__input"
    onChange={[Function]}
    type="checkbox"
  />
  <label
    className="checkbox__text "
  />
</div>
`;
